knotx-core/src/main/java/io/knotx/dataobjects/Knot.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (C) 2016 Cognifide Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.knotx.dataobjects;
+
+import com.google.common.base.Objects;
+import com.google.common.collect.Lists;
+import io.vertx.codegen.annotations.DataObject;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import java.util.List;
+
+@DataObject(inheritConverter = true)
+public class Knot {
```

Comment:
Correct me if I understood this wrong, but this class looks to me more like `KnotTask` or `KnotJob` (or smth. like that).
This is not `Knot` and naming it like that will introduce a lot of confusion. See here: https://github.com/Cognifide/knotx/wiki/Knot what `Knot` is.

If you will apply rename change here, please also think about `KnotStatus` and `KnotError` classes.

Reply:
renamed to KnotTask